lib/index.js

```diff
@@ -140,6 +140,14 @@ function getOptions(options) {
   options.sourceMap = getSourceMap(options);
   options.style = getStyle(options) || 0;
 
+  // context object represents node-sass environment
+  options.context = {
+    info: module.exports.info,
+    state: {},
+    sync: false,
```

Comment:
Thanks for explaining the use-cases. 

* We are deep cloning the option object. 
* Since the user can do `this.options.whatever = { 'bah': 'blah' }` to fulfill their custom requirements, they can append all that to options object the way they please. 
* The idea really is to keep the API itself less noisier and sleeker.